inst/include/mode/solver.hpp

```diff
@@ -106,6 +106,10 @@ class solver {
     t_ = t;
   }
 
+  void set_state(std::vector<double>::const_iterator state) {
+    stepper_.set_state(t_, state);
```

Comment:
should we also reset the step size?

Reply:
I think that this needs to be an option that the user can control:

* If we reset time we _definitely_ should reset the step size
* If we reset the parameters (not yet implemented) we _probably_ should reset the step size
* If we reset the state we _possibly_ should reset the step size

The step size will adjust regardless (possibly through a set of rejected steps), but if the user has made only a small modification to the state we might not expect a large change in step size and the last step might be a better approximation of it.

Similar to the handling of "set initial", I think tidiest to add a `reset_step_size = NULL` option at the R-facing function, then if the user has not added an explicit argument, within your mode_update_state function resolve this based on the presentness of time/parameters/state.

Reply:
Ok, so if we reset *only* time, we should def reset step size, but if we reset time *and* state and the user hasn't specified, we should also default to resetting you think?

Reply:
yes - following the most resetty of the different elements (so of time + anything else is set, and reset_step_size is NULL we reset the step size)